services/spar/src/Spar/Scim/User.hs

```diff
@@ -111,12 +113,31 @@ instance Scim.UserDB Spar where
     updateValidScimUser tokinfo uidText =<< validateScimUser tokinfo newScimUser
 
   delete :: ScimTokenInfo -> Text -> Scim.ScimHandler Spar Bool
-  delete _ _ =
-      throwError $ Scim.ScimError
-          mempty
-          (Scim.Status 404)
-          Nothing
-          (Just "User delete is not implemented yet")  -- TODO
+  delete ScimTokenInfo{stiTeam} uidText = do
+    uid :: UserId <- parseUid uidText
+    mbBrigUser <- lift (Intra.Brig.getUser uid)
+    case mbBrigUser of
+      -- If the user doesn't exist, the deletion likely occurred already.
+      -- to be idempotent we continue assuming the user has been deleted
+      Nothing -> return False
+      Just brigUser -> do
+        unless (userTeam brigUser == Just stiTeam) $ 
+          throwError $ Scim.forbidden "you are not authorized to delete this user"
+        ssoId <- maybe (logThenServerError $ "no userSSOId for user " <> cs uidText)  
+                       pure
+                       $ Brig.userSSOId brigUser
+        uref <- either logThenServerError pure $ Intra.Brig.fromUserSSOId ssoId
+        lift . wrapMonadClient $ Data.deleteSAMLUser uref 
+        lift . wrapMonadClient $ Data.deleteScimUser uid
+        lift (Intra.Brig.deleteUser uid)
```

Comment:
I was initially trying to catch 'not found' errors from all these services to ignore them, but it seems that they're never actually thrown? Maybe brig and cassandra know to just return success if it's already missing?